analyzer/merger-cache-invalidation-plugin/src/main/java/eu/fasten/analyzer/mergercacheinvalidationplugin/MergerCacheInvalidationPlugin.java

```diff
@@ -0,0 +1,188 @@
+            var json = new JSONObject(record);
+            if (json.has("input")) {
+                if (json.get("input").toString().isEmpty()) {
+                    logger.error("Empty input");
+                    setPluginError(new RuntimeException("Empty input"));
+                    return;
+                }
+                json = json.getJSONObject("input");
+            }
+
+            // Parse input values for the root product.
+            String groupId = "";
+            String artifactId = "";
+            String version = "";
+            try {
+                if (!json.has("product")) {
+                    throw new JSONException("product");
+                }
+                if (!json.has("version")) {
+                    throw new JSONException("version");
+                }
+                version = json.get("version").toString();
+                var product = json.get("product").toString();
+                var splits = product.split(Constants.mvnCoordinateSeparator);
+                groupId = splits[0];
+                artifactId = splits[1];
+            } catch (JSONException e) {
+                logger.error("Error parsing product for vulnerability cache invalidator", e);
+                setPluginError(e);
+            }
+
+            // Resolve the set of transitive dependants for this product.
+            this.depSet = this.graphResolver.dependentBFS(groupId, artifactId, version, -1, true);
+
+            // Go over the set and invalidate the cache for each dependant.
+            for (Revision revision : depSet) {
+                var firstLetter = revision.artifactId.substring(0, 1);
+                var outputPath = File.separator + firstLetter +
+                                 File.separator + revision.artifactId +
+                                 File.separator + revision.product().toString() +
+                                 File.separator + revision.version.toString() + ".json";
+
+                File outputFile = new File(outputPath);
+                if(!outputFile.exists() || outputFile.length() == 0) return;
+
+                // TODO: possible optimization: we don't really need to delete the whole file, or do we?
+                var result = outputFile.delete();
+                if(result) {
+                    logger.debug("Successfully invalidated cache for a dependant: " + outputPath);
+                } else {
+                    logger.error("Error in invalidating a dependant cache file: " + outputPath);
+                }
+            }
+        }
+
+        @Override
+        public Optional<String> produce() {
+            if (this.depSet == null) {
+                return Optional.empty();
+            } else {
+                var json = new JSONArray();
+                // TODO: possible optimization: we don't really need to pass the whole revision.
+                var depSetJson = depSet.stream().map(Revision::toJSON);
```

Comment:
I think this and the next line should be substituted with the following to produce the proper JsonArray `depSet.stream().map(Revision::toJSON).forEach(json::put);`